File: crypto/src/main/java/io/oversec/one/crypto/ZxcvbnResult.java
package io.oversec.one.crypto;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by yao on 17/01/17.
 */

public class ZxcvbnResult implements Parcelable {

    private String mWarning;
    private int mEntropy;

    public ZxcvbnResult(Parcel source) {
        mWarning = source.readString();
        mEntropy = source.readInt();
    }

    public ZxcvbnResult(String result, int entropy) {
        mWarning = result;
        mEntropy = entropy;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(mWarning);
        dest.writeInt(mEntropy);
    }

    public static final Creator<ZxcvbnResult> CREATOR = new Creator<ZxcvbnResult>() {
        @Override
        public ZxcvbnResult[] newArray(int size) {
            return new ZxcvbnResult[size];
        }

        @Override
        public ZxcvbnResult createFromParcel(Parcel source) {
            return new ZxcvbnResult(source);
        }
    };

    public CharSequence getWarning() {
        return mWarning;
    }

    public int getEntropy() {
        return mEntropy;
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/ZxcvbnResult.kt
package io.oversec.one.crypto

import android.os.Parcel
import android.os.Parcelable

class ZxcvbnResult : Parcelable {

    private var mWarning: String? = null
    var entropy: Int = 0
        private set

    val warning: CharSequence?
        get() = mWarning

    constructor(source: Parcel) {
        mWarning = source.readString()
        entropy = source.readInt()
    }

    constructor(result: String, entropy: Int) {
        mWarning = result
        this.entropy = entropy
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(mWarning)
        dest.writeInt(entropy)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<ZxcvbnResult> = object : Parcelable.Creator<ZxcvbnResult> {
            override fun newArray(size: Int): Array<ZxcvbnResult?> {
                return arrayOfNulls(size)
            }

            override fun createFromParcel(source: Parcel): ZxcvbnResult {
                return ZxcvbnResult(source)
            }
        }
    }
}
